hoomd/hpmc/pytest/test_shape_utils.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+import numpy as np
+import pytest
+import coxeter
+from plato import geometry
+from pyhull.convex_hull import ConvexHull
+import hoomd
+from hoomd import hpmc
+from copy import deepcopy
+
+
+def get_outward_normal(verts, face, thresh=0.0001):
+    assert (len(face) == 3)
+    (a, b, c) = verts[face]
+    n = np.cross((b - a), (c - a))
+    flip = False
+    for k in range(len(verts)):
+        if (not k in face):
+            d = n.dot(verts[k] - a)
+            if abs(d) > thresh and d > 0:  # by convexity
+                flip = True
+                break
+    if flip:
+        n = -n
+    return n
+
+
+def sort_faces(verts, faces):
+    sorted_face = []
+    for face in faces:
+        nout = get_outward_normal(verts, face)
+        (a, b, c) = verts[face]
+        n = np.cross((b - a), (c - a))
+        if n.dot(nout) > 0:
+            sorted_face.append(face)
+        else:
+            sorted_face.append(face[::-1])
+    return np.array(sorted_face)
+
+
+def _vertices():
+    platonic_shapes = [
+        "Cube", "Tetrahedron", "Octahedron", "Icosahedron", "Dodecahedron"
+    ]
+    verts = []
+    names = []
+    for shape in platonic_shapes:
+        verts.append(
+            np.asarray(
+                coxeter.families.PlatonicFamily.get_shape(shape).vertices))
+        names.append(shape)
+    for i in range(20):
+        verts.append(np.random.rand(np.random.randint(10, 128), 3))
+        names.append("random" + str(i))
+    return zip(names, verts)
+
+
+@pytest.fixture(scope="function", params=_vertices(), ids=(lambda x: x[0]))
```

Comment:
I prefer `@pytest.mark.parameterize` over fixtures like this. See `md/pytest/test_bond.py` for an example (recently updated on the `master` branch): https://github.com/glotzerlab/hoomd-blue/blob/master/hoomd/md/pytest/test_bond.py

Reply:
This is not resolved. I still find the test code very hard to follow. Combined with the suggestion above to hard-code the properties, this code will be very clean and not have any external dependencies:
```
test_parameters = [ ('Cube', [(-1,-1,-1), ...]), ('Tetrahedron', [...]),]
@pytest.mark.parametrize('name, vertices',
                         test_parameters)
def test_convex_hull_vertices(name, vertices):
```

Reply:
I have completely rewritten this test. I'm comparing volume and moment of inertia results against `coxeter` values for various regular and deformed convex polyhedra and ellipsoids, which I have hard coded.

I also removed the tests that checked for convex hull and faces since we switched to using `QuickHull` internally.